reconcile/utils/aws_api.py

```diff
@@ -1209,3 +1212,30 @@ def upsert_route53_record(self, account_name, zone_id, recordset):
                           f'unknown DNS zone {zone_id}')
         except Exception as e:
             logging.error(f'[{account_name}] unhandled exception: {e}')
+
+    def get_image_ids(self, account_name: str, region_name: str,
+                      owners: List[str], filters: Sequence[FilterTypeDef],
+                      name_regex: str,
+                      sort_ascending: bool = False) -> List[str]:
+        """
+        Get a list of AMI IDs matching the specified criteria.
+
+        :param account_name: the account name to operate on
+        :param region_name: aws region name to operate on
+        :param owners: list of AMI owners to limit search.
+        :param filters: One or more name/value pairs to filter off of.
+        https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-images.html
+        :param name_regex: a regex string to apply to the AMI list returned
+                           by AWS.
+        :param sort_ascending: used to sort AMIs by creation time.
+        """
+        ec2 = self._account_ec2_client(account_name, region_name)
+        images = ec2.describe_images(
+            Owners=owners, Filters=filters)['Images']
+        if name_regex:
+            images = [i for i in images if re.match(name_regex, i['Name'])]
+        images = sorted(images,
+                        key=lambda i: datetime.strptime(i['CreationDate'],
```

Comment:
The good news is that the date format you declare (RFC-3339) works perfectly with string comparisons. You can just do
```python
sorted(images, key=lambda i: i['CreationDate'])
```
(unless this is user-controlled input and you want an exception if they pass shit to us).